doc/compiler/control/OptionsProcessing.md

```diff
@@ -0,0 +1,124 @@
+<!--
+Copyright (c) 2023, 2023 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] https://openjdk.org/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Overview
+
+The compiler component options initialization occurs in several stages;
+this doc outlines the order in which the initialization and processing
+occurs, and describes the various stages.
+
+# High Level Outline
+
+1. FrontEnd preprocess
+2. JIT preprocess
+3. Process Options (excluding option sets)
+4. JIT postprocess
+5. FrontEnd postprocess
+6. FrontEnd late postprocess
+7. JIT Late postprocess
+8. Process option sets
+9. JIT late postprocess on option sets
+10. FrontEnd late postprocess on option sets
+
+# Details
+
+Since the compiler processes both `-Xjit` and `-Xaot`, both of which
+are represented by `TR::Options`, the initialization/processing
+occurs twice. `onLoadInternal` (which is called when the JVM is in the
+`JIT_INITIALIZED` init phase) is where the global AOT and JIT
+`TR::Options` objects are allocated, initialized, and processed.
+`aboutToBootstrap` (which is called when the JVM is in the
+`ABOUT_TO_BOOTSTRAP` init phase) is where late post processing occurs,
+as well as the processing of option sets.
```

Comment:
I wonder if it'd be good to explicitly mention the two phases and also put the two phases in bulletin points. 
 
```
Since the compiler processes both `-Xjit` and `-Xaot`, both of which
are represented by `TR::Options`, the initialization/processing
occurs twice in two initialization phases. 

1.  `onLoadInternal`
     - Called when the JVM is in the `JIT_INITIALIZED` init phase
     - The is where the global AOT and JIT `TR::Options` objects are allocated, initialized, and processed.
2.  `aboutToBootstrap`
     - Called when the JVM is in the`ABOUT_TO_BOOTSTRAP` init phase
     - This is where late post processing occurs, as well as the processing of option sets.
```

Reply:
Done in force push.